hotel/models/hotel_folio.py

```diff
@@ -0,0 +1,810 @@
+        return folio_id
+
+    def write(self, vals):
+        """
+        Overrides orm write method.
+        @param self: The object pointer
+        @param vals: dictionary of fields value.
+        """
+        product_obj = self.env["product.product"]
+        h_room_obj = self.env["hotel.room"]
+        folio_room_line_obj = self.env["folio.room.line"]
+        room_lst = []
+        room_lst1 = []
+        for rec in self:
+            for res in rec.room_lines:
+                room_lst1.append(res.product_id.id)
+            if vals and vals.get("duration_dummy", False):
+                vals["duration"] = vals.get("duration_dummy", 0.0)
+            else:
+                vals["duration"] = rec.duration
+            for folio_rec in rec.room_lines:
+                room_lst.append(folio_rec.product_id.id)
+            new_rooms = set(room_lst).difference(set(room_lst1))
+            if len(list(new_rooms)) != 0:
+                room_list = product_obj.browse(list(new_rooms))
+                for rm in room_list:
+                    room_obj = h_room_obj.search([("name", "=", rm.name)])
+                    room_obj.write({"isroom": False})
+                    vals = {
+                        "room_id": room_obj.id,
+                        "check_in": rec.checkin_date,
+                        "check_out": rec.checkout_date,
+                        "folio_id": rec.id,
+                    }
+                    folio_room_line_obj.create(vals)
+            if len(list(new_rooms)) == 0:
+                room_list_obj = product_obj.browse(room_lst1)
+                for rom in room_list_obj:
+                    room_obj = h_room_obj.search([("name", "=", rom.name)])
+                    room_obj.write({"isroom": False})
+                    room_vals = {
+                        "room_id": room_obj.id,
+                        "check_in": rec.checkin_date,
+                        "check_out": rec.checkout_date,
+                        "folio_id": rec.id,
+                    }
+                    folio_romline_rec = folio_room_line_obj.search(
+                        [("folio_id", "=", rec.id)]
+                    )
+                    folio_romline_rec.write(room_vals)
+        return super(HotelFolio, self).write(vals)
+
+    @api.onchange("partner_id")
+    def onchange_partner_id(self):
+        """
+        When you change partner_id it will update the partner_invoice_id,
+        partner_shipping_id and pricelist_id of the hotel folio as well
+        ---------------------------------------------------------------
+        @param self: object pointer
+        """
+        if self.partner_id:
+            order_ids = [folio.order_id.id for folio in self]
+            self.update(
+                {
+                    "partner_invoice_id": self.partner_id.id,
+                    "partner_shipping_id": self.partner_id.id,
+                    "pricelist_id": self.partner_id.property_product_pricelist.id,
```

Comment:
These fields are required when creating an invoice, should thy be marked as compulsory?

Reply:
These 3 fields are already defined as required based on the value of the state.